Tools/peg_generator/scripts/test_parse_directory.py

```diff
@@ -187,25 +158,33 @@ def parse_directory(
                 break
 
         if not should_exclude_file:
+            with tokenize.open(file) as f:
+                source = f.read()
             try:
                 if tree_arg:
                     mode = 1
```

Comment:
This modifies a parameter based on another parameter. We should just switch to `mode=1` in the caller if the `tree` arg is True. (An the function should assert that if `tree_arg` is set, `mode` is 1.)